crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -487,14 +505,33 @@ impl DiskFileSystem {
             path.join(&*unix_to_sys(&fs_path.path))
         })
     }
+
+    fn invalidate_from_write(&self, full_path: &Path, invalidators: HashSet<Invalidator>) {
+        if !invalidators.is_empty() {
+            if let Some(path) = format_absolute_fs_path(full_path, &self.name, self.root_path()) {
+                if invalidators.len() == 1 {
+                    let invalidator = invalidators.into_iter().next().unwrap();
+                    invalidator.invalidate_with_reason(Write { path });
```

Comment:
Nit: I don't think we need to optimize this path given that it's here just to prevent us from ever hitting this path.

Reply:
old_invalidators can also contain invalidators from `read` methods. It's not `write`-only.

So having exactly 1 `read` invalidator for a file is pretty likely.